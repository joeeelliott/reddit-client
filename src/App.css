.Header_header {
  /* width: 90%; */
  /* margin-left: 10%; */
  height: 80px;
  background-color: white;
  /* border: 1px solid blue; */
  display: flex;
  align-items: center;

  /* offset-x | offset-y | blur-radius | color */
  box-shadow: 5px 5px 5px #ddd;

  position: relative; 
  z-index: 10; 
}

.Header_toggle-btn-container {
  /* border: 1px solid green; */
  /* top: 0px; */
  /* left: -11%; */
  height: 100%;
  width: 20%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.Header_logo-title-container {
  /* border: 1px solid red; */
  height: 100%;
  width: 80%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.Header_bars-icon {
  font-size: 1.5rem !important;
  cursor: pointer;
  transition: all 1s ease;
}

.Header_bars-icon-rotate {
  transform: rotate(90deg); 
}

.Header_reddit-icon {
  /* color: lightgrey; */
}

.Header_font-awesome-icon {
  font-size: 2rem;
}

.Header_title {
  font-size: 1.5rem;
}

.Header_title-span {
  color: grey
}

.Header_side-nav {
  position: absolute; 
  /* left: -170px;  */
  left: 0;
  top: 0; 
  height: 100vh; 
  margin-top: 80px;
  /* border: 1px solid red; */
  z-index: -10; 
  font-size: 0.7rem;
  /* width: 170px; */
  width: 0px;
  overflow: hidden;
  background-color: #ddd;
  transition: width 0.5s;
}

.Header_show-side-nav {
  position: absolute; 
  left: 0;
  top: 0; 
  height: 100vh; 
  margin-top: 80px;
  /* border: 1px solid red; */
  z-index: -10; 
  font-size: 0.7rem;
  width: 170px;
  overflow: hidden; 
  background-color: #ddd;
  transition: width 0.5s;
}

.Header_side-nav label, .Header_show-side-nav label {
  display: block; 
  margin: 10px; 
}

.Header_side-nav input, select, .Header_show-side-nav input, select {
  font-size: 0.7rem;
  margin: 0px 10px;
}

.Header_side-nav-btn-container {
  display: flex;
  justify-content: flex-end;
  margin: 20px 10px; 
}

.Header_side-nav-btn {
  font-size: 0.7rem;
  cursor: pointer;
  border: 1px solid grey; 
  border-radius: 20px;
}

/* NOTE TO SELF ----> below is how you toggle rotate on each click  */
/* #shape  { width: 100px; 
  height: 200px; 
  background:#000;
  -moz-transition: all 1s ease;
  -webkit-transition: all 1s ease;
  -o-transition: all 1s ease;
   transition: all 1s ease;
 }

.rotate {-moz-transform: rotate(90deg);
 -webkit-transform:rotate(90deg); 
 -ms-transform:rotate(90deg);
 -o-transform:rotate(90deg);
 } */

/* @media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
} */

/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */
