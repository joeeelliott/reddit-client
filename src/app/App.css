/* APP COMPONENT */


.App {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  overflow: hidden;
  background-color: rgba(120, 160, 187, 0.2);
}

.App_loading-wrapper {
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  background-color: #242f3f;
  display: flex; 
  justify-content: center;
  align-items: center;
  /* display: none; */
}

.loader {
  display: inline-block;
  width: 30px;
  height: 30px;
  position: relative;
  border: 4px solid #fff;
  animation: loader 2s infinite ease; 
}

.loader-inner {
  vertical-align: top; 
  display: inline-block;
  width: 100%;
  background-color: #fff;
  animation: loader-inner 2s infinite ease-in; 
}

@keyframes loader {
  0% {
    transform: rotate(0deg);
  }

  25% {
    transform: rotate(180deg);
  }

  50% {
    transform: rotate(180deg);
  }

  75% {
    transform: rotate(360deg); 
  }

  100% {
    transform: rotate(360deg);
  }
}

@keyframes loader-inner {
  0% {
    height: 0%;
  }

  25% {
    height: 0%;
  }

  50% {
    height: 100%;
  }

  75% {
    height: 100%;
  }

  100% {
    height: 0%;
  }
}

/* .App_header-container {
  position: fixed;
  width: 100%;
  top: 0px;
} */

/* HEADER COMPONENT */


.header_header {
  /* border: 1px solid blue; */
  height: 80px;
  background-color: white;
  display: flex;
  flex-direction: column;
  /* align-items: center; */

  justify-content: center;
  /* offset-x | offset-y | blur-radius | color */
  box-shadow: 5px 5px 5px #ddd;

  position: relative; 
  z-index: 100; 

  position: fixed;
  width: 100%;
  top: 0px;
}

.header_toggle-and-title-container {
  /* border: 1px solid blue; */
  display: flex;
  height: 75%;
  position: relative;
}

.header_toggle-btn-container {
  /* border: 1px solid green; */

  position: absolute;
  left: 0px;
  height: 100%;
  width: 10%;
  display: flex;
  justify-content: center;
  align-items: center;

  z-index: 1; /* so event handler works with absolute positioning */
}

.header_logo-title-container {
  /* border: 1px solid red; */
  /* height: 100%; */
  /* width: 80%; */
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  /* margin-right: 20%;  so the logo and title are in the middle of the header, counters the toggle icon 20% width, didn't want toggle icon in absolute positioning  */
}

.header_logo-title-link {
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
}

.header_search-icon {
  cursor: pointer;
  /* font-size: 1.5rem !important; */
  transition: all 0.5s ease;  /* part 1 of the burger menu rotation */
}

.header_filter-icon {
  font-size: 0.8rem !important;
  cursor: pointer;
  /* transition: all 0.5s ease; */
   /* part 1 of the burger menu rotation */
}

.header_search-icon-rotate {
  transform: rotate(90deg); /* part 2 of the burger menu rotation */
}

.header_reddit-icon {
  color: rgb(255, 69, 0);
}

.header_font-awesome-icon {
  font-size: 1.5rem;
}

.header_title {
  font-size: 1.5rem;
  color: black;
}

.header_title-span {
  color: grey
}

.header_nav-container {
  /* border: 1px solid green; */
  height: 25%;
  width: 100%;
  display: flex;
  align-content: center;
  justify-content: center;
  /* margin-top: 10px; */
}


/* NAV COMPONENT */


.nav_list {
  /* border: 1px solid yellow; */
  display: flex;
  /* justify-content: center; */
  /* align-items: center; */
  list-style: none;
  height: 100%;
}

.nav_link {
  text-decoration: none;
  margin: 0px 10px;
  /* color: black; */
  height: 100%;
  line-height: 100%;
  color: rgb(150, 150, 150);
  /* border: 1px solid orange; */
}

.nav_link:hover {
  color: black;
}

.nav_saved-article-animation {
  animation: saveClicked 1s;
}

@keyframes saveClicked {
  0% {color:rgb(150, 150, 150);}
  25% {color:rgba(144, 238, 144, 0.5);}
  50% {color:rgba(144, 238, 144, 0.5);}
  75% {color:rgba(144, 238, 144, 0.5);}
  100% {color:rgb(150, 150, 150);}
}


/* SIDENAV COMPONENT */

.sideNav_sideNav {
  position: absolute; 
  left: 0;
  top: 0; 
  height: calc(100vh - 80px);
  margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem;
  /* width: 170px; */
  width: 0px;
  overflow: hidden;
  background-color: #ddd;
  border-top-right-radius: 20px;
  border-bottom-right-radius: 20px;
}

.sideNav_hide-nav {
  /* border: 1px solid red; */
  /* position: absolute; 
  left: 0;
  top: 0;  */
  /* height: 100vh;  */
  /* margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem; */
  width: 0px;
  /* overflow: hidden;
  background-color: #ddd; */
  transition: width 0.4s;
}

.sideNav_show-nav {
  /* border: 1px solid red; */
  /* position: absolute; 
  left: 0;
  top: 0;  */
  /* height: calc(100vh - 80px); */
  /* margin-top: 80px;
  z-index: -10; 
  font-size: 0.7rem; */
  width: 170px;
  /* overflow: hidden; 
  background-color: #ddd; */
  transition: width 0.5s;
  /* border-top-right-radius: 20px;
  border-bottom-right-radius: 20px; */
}

.sideNav_hide-nav label, .sideNav_show-nav label {
  display: block; 
  margin: 10px; 
}

.sideNav_hide-nav input, select, .sideNav_show-nav input, select {
  font-size: 0.7rem;
  margin: 0px 10px;
}

.sideNav_btn-container {
  display: flex;
  justify-content: flex-end;
  margin: 20px 10px; 
}

.sideNav_btn {
  font-size: 0.7rem;
  cursor: pointer;
  border: 1px solid grey; 
  border-radius: 20px;
}

.sideNav_eye-text-container {
  /* border: 1px solid blue; */
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
}

.sideNav_eye-icon-container {
  /* border: 1px solid green; */
  display: flex;
  align-items: center;
}

.sideNav_eye-icon {
  font-size: 1rem;
  margin-right: 20px;
  margin-left: 10px;
  cursor: pointer;
}

.sideNav_text-container {
  /* border: 1px solid red; */
}

.sideNav_eye-icon-hover-text {
  font-size: 0.7rem;
  color: rgba(0, 0, 0, 0.5); 
  opacity: 0;
}

.sideNav_eye-icon-hover-text-show {
  animation: showText 2s; 
}

@keyframes showText {
  from {opacity: 0;}
  to {opacity: 1.0;}
}





/* ARTICLE COMPONENT */

.article_container {
  /* border: black 1px solid; */
  border-radius: 15px;
  margin: 100px 13%;  /* make % smaller the wider you want the container. 100px top to fall below the header, 100 from bottom to be able to see the ellipsis dropdown on bottom article */
}

.article_saved-articles-none-saved {
  /* border: 1px solid black; */
  width: 100%; 
  height: calc(100vh - 220px);  /* 100vh take the footer, header, and margins/padding inbetween - didnt actually work out just kept doing trial and error and saw how it changed on the page */ 
  display: flex;
  flex-direction: column; 
  /* align-items: center; */
  justify-content: center;
}

.article_saved-articles-none-saved * {
  text-align: center;
  margin: 10px 0px;
}

.article_saved-articles {
  width: 100%; 
  height: calc(100vh - 220px);  /* 100vh take the footer, header, and margins/padding inbetween - didnt actually work out just kept doing trial and error and saw how it changed on the page */ 
}

.article_saved-articles-2plus {
  width: 100%;
  height: 100%;
}

.article_outer-div {
  position: relative;
}

.article_outer-container {
  /* border: red 1px solid !important; */
  width: 100%;
  display: flex;
  padding: 5px 0px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  border: 1px solid #ccc;
  border-radius: 5px;
}

.article_outer-container-saved-on-render {
  /* border: red 1px solid !important; */
  width: 100%;
  height: 100%;
  display: flex;
  padding: 5px 0px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  border: 1px solid #ccc;
  border-radius: 5px;
}

.article_outer-container-hide {
  display: none;
}

.article_score-container {
  /* border: 1px solid green; */
  flex: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  padding-top: 10px;
}

.article_score {
  /* border: 1px solid lightblue; */
  margin: 10px 0px;
  font-weight: 600;
  display: flex;
  justify-content: center;
  align-items: center;
}

.article_score-icon, .article_comments-icon, .article_share-icon, .article_ellipsis-icon {
  color:rgb(150, 150, 150);
  cursor: pointer; 
}

.article_score-up-icon-clicked {
  color: green; 
}

.article_score-down-icon-clicked {
  color: red;
}

.article_main-content-container {
  /* border: 1px solid darkblue; */
  flex: 12;
  display: flex;
  flex-direction: column;
}

.article_details-top-container {
  /* border: 1px solid green; */
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.article_post-details {
  /* border: 1px solid purple; */
  display: flex;
  align-items: center;
  padding-left: 10px;
}

.article_api-data {
  /* border: 1px solid red; */
  color:rgb(150, 150, 150);
  font-size: 0.8rem;
  text-align: center;
}

.strong {
  color: black !important; 
  font-weight: 600;
  margin-right: 5px;
}

.article_join-btn {
  background-color:rgb(31, 144, 231);
  color: #fff;
  border-radius: 20px;
  padding: 3px 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 10px;
  cursor: pointer;
}

.article_join-btn-text {
  margin: 0px 5px;
  font-size: 0.8rem;
}

.article_title-bottom-details-img-container {
  /* border: 1px solid red; */
  display: flex; 
  justify-content: space-between;
  height: 100%;
}

.article_title-bottom-details-container {
  /* border: 1px solid blue; */
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.article_title-container {
  /* border: 1px solid red; */
  padding-left: 10px;
}

.article_title {
  /* border: 1px solid green; */
  font-size: 1rem;
  font-weight: 500;
}

.article_bottom-details-container {
  /* border: 2px solid yellow; */
  display: flex;
  flex-direction: row;
  align-items: center;
  font-size: 0.8rem;
  margin-top: 10px;
  padding-left: 10px;
}

.article_comments-icon-container, .article_share-icon-container {
  display: flex;
  align-items: center;
  padding: 0px 4px;
  cursor: pointer;
}

.article_share-icon-container {
  margin-left: 25px;
}

.article_comments-icon-container:hover, .article_share-icon-container:hover {
  background-color:rgba(120, 160, 187, 0.2)
}

.article_comments-icon {
  margin-right: 5px;
}

.article_share-icon {
  margin-right: 5px;
}

.article_ellipsis-container {
  margin-left: 25px;
  position: relative;
  /* border: 1px solid black; */
  /* background-color: green; */
  padding: 0px 4px;
}

.article_ellipsis-container:hover {
  background-color:rgba(120, 160, 187, 0.2)
}

.article_ellipsis-icon {
  /* margin-left: 25px; */
  font-weight: normal;
  cursor: pointer;
}

.article_ellipsis-dropdown {
  /* border: 1px solid green; */
  /* height: 60px;
  width: 10px; */
  position: absolute;
  left: 0px;
  bottom: -80px;
  background-color: rgb(248, 248, 240); 
  /* background-color: red; */
  display: none;
  z-index: 55;
}

.article_ellipsis-dropdown-show {
  display: block !important; 
}

.article_save-container, .article_hide-container, .article_report-container {
  /* border: 1px solid red; */
  width: 100px;
  height: 25px;
  display: flex;
  flex-direction: row;
  /* justify-content: space-between; */
  cursor: pointer; 
  color: rgba(0, 0, 0, 0.5); 
  background-color: rgb(248, 248, 240); 
}

.article_save-container-clicked, .article_hide-container-clicked, .article_report-container-clicked {
  width: 100px;
  height: 25px;
  display: flex;
  flex-direction: row;
  /* justify-content: space-between; */
  cursor: pointer; 
  color: rgba(0, 0, 0, 0.5);
  background-color: rgba(144, 238, 144, 0.5); 
}

.article_report-container-clicked {
  background-color: rgba(255, 0, 0, 0.8) !important; 
}

.article_save-container:hover, .article_hide-container:hover, .article_report-container:hover {
  background-color: rgb(230, 230, 230); 
  color: #000; 
}

.article_save-container-clicked:hover, .article_hide-container-clicked:hover {
  background-color: rgba(144, 238, 144, 0.5);  
  color: #000; 
}

.article_report-container-clicked:hover {
  color: #000;
}

.article_save-icon-container, .article_hide-icon-container, .article_report-icon-container {
  /* border: 1px solid blue;  */
  width: 40%;
  /* width: 80px;
  height: 25px;
  display: flex;
  flex-direction: row; */
  display: flex;
  align-items: center;
  justify-content: center;
  /* text-align: center; */
}

.article_save-icon, .article_hide-icon, .article_report-icon {
  /* margin-right: 5px; */
  /* flex: 2; */
  /* height: 100%; */
  /* font-size: 0.5rem; */
  font-size: 0.9rem;
  width: 30px;
  /* font-weight: 0.8rem; */
}

.article_save-text, .article_hide-text, .article_report-text {
  /* border: 1px solid black; */
  align-self: center;
  display: flex;
  /* justify-content: center; */  
  /* font-size: 0.9rem; */
}

.article_save-text, .article_hide-text, .article_report-text {
  /* flex: 1; */
  width: 50px;
}

.article_report-text:hover {
  /* background-color: rgb(230, 230, 230); 
  color: #000;  */
}

.article_save-icon-container {
  /* border: 1px solid red; */
}

.article_save-icon {
  /* display: inline-block; */
  /* border: 1px solid black; */
}

.article_hide-icon-container {
  /* border: 1px solid blue; */
}

.article_hide-icon {
  /* border: 1px solid black; */
  /* font-size: 1rem; */
}

.article_report-icon-container {
  /* border: 1px solid yellow; */
}

.article_report-icon {
  /* border: 1px solid black; */
}

.article_img-container {
  /* border: 2px solid orange; */
  padding: 5px;
  display: flex;
  align-items: center;
  position: relative;
}

.article_article-img {
  /* border: 1px solid rgb(0, 121, 211); */
  cursor: pointer;
}

.article_article-img-clicked {
  /* border: 1px solid rgb(0, 121, 211); */
  border-radius: 5px;
}

.article_img-modal {
  position: absolute; 
  border: 1px solid rgba(150, 150, 150);
  border-radius: 5px;
  height: auto;
  width: auto;
  top: 0px;
  right: 0px;
  padding: 20%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: rgba(233, 233, 233, 0.9);
  z-index: 50;
}

.article_img-modal-close-icon {
  position: absolute;
  top: 3%;
  right: 3%;
  color: rgba(150, 150, 150);
  cursor: pointer;
}

.article_img-modal-close-icon:hover {
  color: red;
}

.article_report-modal-hidden {
  position: fixed; 
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 100; 
  background-color:rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  /* display: none; */
  visibility: hidden;  
}

.article_report-modal {
  position: fixed; 
  top: 0;
  left: 0;
  height: 100vh;
  width: 100vw;
  z-index: 100; 
  background-color:rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  /* display: none; */
  /* visibility: hidden;  */
}

.article_report-modal-content {
  height: 50%;
  width: 50%;
  background-color: rgba(255, 255, 255, 0.9); 
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  text-align: center;
  border-radius: 2%;
  padding: 20px 0px;
}

.article_report-modal-report-icon {
  font-size: 2rem;
  color: red;
}

.article_report-modal-btn-container button {
  padding: 2px;
  background-color: rgba(230, 230, 230);
  border: 1px solid grey;
  border-radius: 5px;
}

.article_report-modal-btn-container button:hover {
  background-color: rgba(0, 0, 0);
  color: #fff;
}


/* FOOTER COMPONENT */


.footer {
  /* position: fixed; */
  /* bottom: 0px; */
  /* border: 1px solid black; */
  width: 100%;
  height: 5%;
  box-shadow: 0px -15px 20px #ddd;  
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
}

.footer h1 {
  font-size: 0.8rem;
  margin: 0px 2px;
  /* border: 1px solid black; */
}

.footer p {
  font-size: 0.8rem;
  margin: 0px 2px;
}



/* NOTE TO SELF ----> below is how you toggle rotate on each click  */
/* #shape  { width: 100px; 
  height: 200px; 
  background:#000;
  -moz-transition: all 1s ease;      <------- transition property
  -webkit-transition: all 1s ease;
  -o-transition: all 1s ease;
   transition: all 1s ease;
 }

.rotate {-moz-transform: rotate(90deg);   <------ rotate property on 
 -webkit-transform:rotate(90deg);                 a class that will be 
 -ms-transform:rotate(90deg);                     toggled 
 -o-transform:rotate(90deg);
 } */


 
/*       below is example of an infinite rotation animation 
@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
} */

/* @keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
} */

