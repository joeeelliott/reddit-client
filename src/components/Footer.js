import React from 'react';

const Footer = () => {
  return (
    <div data-testid="footer" className="footer">
      <h1>Joe Elliott</h1>
      <p>&copy;</p>
    </div>
  )
}

export default Footer;