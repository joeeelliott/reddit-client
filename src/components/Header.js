import React from 'react';
import SideNav from './SideNav'; 
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useSelector, useDispatch } from 'react-redux'; 
import { selectShowSideNav, showSideNav } from '../features/sideNavSlice';

const Header = () => {
  const sideNavState = useSelector(selectShowSideNav)
  const dispatch = useDispatch();

  const handleShowSideNavClick = (e) => {
    sideNavState.toggleSideNav ? e.currentTarget.classList.toggle('Header_search-icon-rotate') : e.currentTarget.classList.toggle('Header_search-icon-rotate'); 

    dispatch(showSideNav()); 
  }

  return (
    <div className="Header_header">
      <div className="Header_toggle-btn-container">
        <FontAwesomeIcon icon="search" data-test-Header_search-icon data-test-Header_font-awesome-icon onClick={handleShowSideNavClick} />
      </div>

      <div className="Header_logo-title-container">
        <FontAwesomeIcon icon={['fab', 'reddit']} className="Header_font-awesome-icon" />

        <h1 className="Header_title">Reddit<span className="Header_title-span">Sample</span></h1>
      </div>

      <SideNav />
    </div>
  )
}

export default Header;