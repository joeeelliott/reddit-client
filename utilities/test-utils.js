// THIS FILE IS FOR REFACTORED TEST FUNCTIONS THAT WILL BE KEPT IN A GLOBAL STATE AS THEY ARE LIKELY TO BE USED ACROSS MULTIPLE FILES

// See https://redux.js.org/usage/writing-tests for more info on below code. May be useful at a later date when using integration testing API calls with fetch, msw etc. This is setup, other code on there for example of the integration test. 

import React from 'react'; 
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom'; 
// Import your own reducer
import sideNavReducer from '../src/features/sideNavSlice';
import postReducer from '../src/features/postSlice'; 

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { sideNav: sideNavReducer, posts: postReducer }, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return ( 
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    )
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
//override render method
export { render }
