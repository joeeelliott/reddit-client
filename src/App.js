import './App.css';

import Header from './components/header/Header'; 

function App() {
  return (
    <div data-test="App">
      <Header />
    </div>
  );
}

export default App;
