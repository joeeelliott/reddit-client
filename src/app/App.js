import './App.css';

import Header from '../components/Header'; 
import Article from '../components/Article';
import Footer from '../components/Footer'; 

function App() {
  // <script src="jquery-2.1.4.js"></script>

  return (
    <div data-test="App">
      <div className="App_loading-wrapper">
        <span class="loader"><span class="loader-inner"></span></span>
      </div>
      
      <Header />
      <div className="article_container">
        <Article />
      </div>
      <Footer />

      {/* <script>
        $(window).on('load', function() {
          $('.loader-wrapper').fadeOut('slow')
        });
      </script> */}  
      
      {/* {need to find a way to load this in on window load - maybe in article slice will pop as no display for now but thats not currently pushed and merged to master yet so will do that later} */}
    </div>
  );
}

export default App;
