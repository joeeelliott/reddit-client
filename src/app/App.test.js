import App from './App';
import Header from '../components/Header';
import Footer from '../components/Footer';

import { shallow } from 'enzyme';  // creates an instance of your component

// import { mount } from 'enzyme';
// Shallow wrapping doesnâ€™t descend down to sub-components. A full mount also mounts sub-components. We use just shallow here

import { findByTestAttr } from '../../utilities/test-utils'; 

// REFACTORING FUNCTIONS TO PREVENT REPETITIVE CODE ---- 

// any kept in here and not in utilities folder is because they are likely to change slightly from file to file, but will be used repetitively within that file so best to write individually

// Creates a shallow render wrap for the <App /> component 
const shallowRender = (props={}) => {
  const appWrapper = shallow(<App {...props} />); 
  return appWrapper; 
}


// describe creates a test suite 
describe('App Component', () => {

  // Prevents us having to declare shallow render variable in each test by invoking the shallowRender before each test
  let appWrapper;
  beforeEach(() => {
    appWrapper = shallowRender();
  }); 

  describe('Rendering of elements', () => {

    // it('App component renders', () => {
    //   // console.log(appWrapper.debug());
    //   expect(appWrapper).toBeDefined();
    // });

    it('App component renders', () => {
      // console.log(appWrapper.debug());
      expect(appWrapper).toHaveLength(1);
    }); 
  
  
    // -------------------------------------------------------
    // .find('element/component').toHaveLength(num of times expected to appear)    ---->  IS BEST PRACTICE FOR TESTING RENDERING OF ELEMENTS/COMPONENTS. 
    // -------------------------------------------------------
    

    // VALID
    it('renders one <Header /> component', () => {
      // console.log(appWrapper.debug());
      expect(appWrapper.find('Header')).toHaveLength(1);
    });  // components with capital, elements with lower case
  
    // VALID
    it('renders one <div> component', () => {
      // console.log(appWrapper.debug());
      // expect(appWrapper.find('div').length).toBe(1);
      expect(appWrapper.find('div')).toHaveLength(1);
      // either one of above works 
    });
  
    it('renders in all elements', () => {
      expect(appWrapper.containsMatchingElement(
        <div>
          <Header />
          <Footer />
        </div>
      )).toBeTruthy();
    }); // can test single or multiple elements, but if multiple, anything inside the most outer tags you're looking for has to match EXACTLY as it is in file (except attributes). E.g. if the <Header /> wasn't included in this test set up and it was just searching for <div></div>, it would fail. 


    // USING DATASETS OVER CLASSES WARNS OTHER DEVELOPERS TO NOT CHANGE THEM AS THEY ARE USED FOR TESTING PURPOSES. IF PART OF A TEST, CHANGE FROM CLASS TO DATA-SET. 

    // found by className
    // it('has a div with a class of "App"', () => {
    //   const app = appWrapper.find('.App'); 
    //   expect(app.length).toBe(1);
    // });

    // found by data-set
    it('has a div with a data-test of "App"', () => {
      // const div = appWrapper.find(`[data-test='App']`);

      // the above code refactored...
      const div = findByTestAttr(appWrapper, 'App')
      expect(div.length).toBe(1);  
    });
  });
});