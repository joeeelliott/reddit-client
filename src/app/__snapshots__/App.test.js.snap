// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders as expected; matches snapshot 1`] = `ShallowWrapper {}`;
